### PR TITLE
157 review eq operators

### PR DESCRIPTION
**Pull request by vo-nil**

### CODE REVIEW

File: include/nil/crypto3/algebra/curves/detail/edwards/element_g2.hpp [hunk 1]
```diff
@@ -277,27 +281,28 @@ namespace nil {
...
+                            return *this;
                         }
+
                         /** @brief
                          *
                          * “Mixed addition” refers to the case Z2 known to be 1.
                          * @return addition of two elements from group G2
                          */
-                        constexpr element_edwards_g2 mixed_add(const element_edwards_g2 &other) const {
+                        constexpr element_edwards_g2& mixed_add(const element_edwards_g2 &other) {
```
Comment (martun) [inline include/nil/crypto3/algebra/curves/detail/edwards/element_g2.hpp:297]: Can we return "void" here?

File: include/nil/crypto3/algebra/curves/detail/forms/edwards/inverted/add_2007_bl.hpp [hunk 2]
```diff
@@ -41,7 +41,7 @@ namespace nil {
                     struct edwards_element_g1_inverted_add_2007_bl {
 
                         template<typename ElementType>
-                        constexpr static inline ElementType process(const ElementType &first,
+                        constexpr static inline ElementType& process(ElementType &first,
```
[the remark this answers is not in the document]
Comment (vo-nil) [reply include/nil/crypto3/algebra/curves/detail/forms/edwards/inverted/add_2007_bl.hpp]: updated. now all operations return void

File: include/nil/crypto3/algebra/curves/detail/forms/montgomery/element_g1_affine.hpp [hunk 3]
```diff
@@ -212,19 +212,24 @@ namespace nil {
                         }
 
                         constexpr curve_element operator+(const curve_element &other) const {
-                            if (this->is_zero()) {
```
Comment (martun) [inline include/nil/crypto3/algebra/curves/detail/forms/montgomery/element_g1_affine.hpp:215]: Why did you change this part?

File: include/nil/crypto3/algebra/curves/detail/forms/short_weierstrass/element_g1_affine.hpp [hunk 4]
```diff
@@ -183,20 +183,24 @@ namespace nil {
                         }
 
                         constexpr curve_element operator+(const curve_element &other) const {
+                            curve_element result = *this;
                             // handle special cases having to do with O
-                            if (this->is_zero()) {
```
Comment (martun) [inline include/nil/crypto3/algebra/curves/detail/forms/short_weierstrass/element_g1_affine.hpp:187]: Same here, this looked good, you can create the result variable lower.

File: include/nil/crypto3/algebra/curves/detail/forms/montgomery/element_g1_affine.hpp (hunk 3 above)
Comment (vo-nil) [reply include/nil/crypto3/algebra/curves/detail/forms/montgomery/element_g1_affine.hpp]: updated

**Review by martun — APPROVED** [review]
